### PR TITLE
MVP: Rust Websocket for deployer with React SPA for variable discovery and frontent

### PR DESCRIPTION
**Pull request by lawrencegripper**
Here is my first shot a rust exe which will start a websocket server. When it receives a message of "start" it will run terraform and stream back the resulting stdout over the web socket. 

Todos:

- [x] Tests
- [x] Check stderr is captured
- [x] Allow client to send back responses to requests for input
- [ ] Add AES encryption to message passing (with pre-shared key)

### CODE REVIEW

**Comment by lawrencegripper** [comment]
I've integrated the rust based deployer with @jjcollinge 's work and have a basic MVP. 

end2end-integrated

Things that are not quite right:

- The web client sometimes experiences issues connecting to the deployer. These are usually transient and I haven't worked out the cause yet. 
- The deployer and client don't yet use any crypto

I think we should merge this into master as the repo is still WIP then we can branch off this for further changes.

**Review by jjcollinge — CHANGES_REQUESTED** [review]
Resolve conflicts and we'll merge 👍

File: tfdeploy/src/components/preview.jsx [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+                await sleep(8000);
+                await this.connect(ipAddress, e => {
+                    xterm.writeln(forcedChalk.red(`\r\n\r\n Connection closed ${skull}, giving up`));
+                });
+                clearInterval(t);
+            });
+
+            clearInterval(t);
+
+            xterm.writeln(forcedChalk.greenBright("Connected interactive terminal to Terroform container \n\n"));
```
Comment (jjcollinge) [inline tfdeploy/src/components/preview.jsx]: Terroform -> Terraform

**Review by jjcollinge — APPROVED** [review]
LGTM

**Pull request merged**